Spigot-Server-Patches/0611-Add-warning-for-servers-not-running-on-Java-11.patch

```diff
@@ -4,19 +4,24 @@ Date: Wed, 2 Dec 2020 21:58:45 -0800
 Subject: [PATCH] Add warning for servers not running on Java 11
 
 
-diff --git a/src/main/java/com/destroystokyo/paper/util/PaperJvmChecker.java b/src/main/java/com/destroystokyo/paper/util/PaperJvmChecker.java
+diff --git a/src/main/java/io/papermc/paper/util/PaperJvmChecker.java b/src/main/java/io/papermc/paper/util/PaperJvmChecker.java
 new file mode 100644
-index 0000000000000000000000000000000000000000..afe07487b1e7337293f7147619a6f10fcdef9d0e
+index 0000000000000000000000000000000000000000..d761a9e0c1e71fadb981f9af4736d7e8e6fda2a9
 --- /dev/null
-+++ b/src/main/java/com/destroystokyo/paper/util/PaperJvmChecker.java
-@@ -0,0 +1,49 @@
-+package com.destroystokyo.paper.util;
++++ b/src/main/java/io/papermc/paper/util/PaperJvmChecker.java
+@@ -0,0 +1,58 @@
++package io.papermc.paper.util;
 +
 +import org.apache.logging.log4j.LogManager;
 +import org.apache.logging.log4j.Logger;
 +
++import java.util.regex.Matcher;
++import java.util.regex.Pattern;
++
 +public class PaperJvmChecker {
 +
++    private final static Pattern NUM_PATTERN = Pattern.compile("\\d+");
```

Comment:
We might as well use regex for handling the `1.` part too. Let's do:
```
(?:1\.)?(\d+)
```
And an error message for when the pattern fails would be good too.

So that would simplify the whole method to:
```java
private static int getJvmVersion() {
    String javaVersion = System.getProperty("java.version");
    final Matcher matcher = Pattern.compile("(?:1\\.)?(\\d+)").matcher(javaVersion);
    if (!matcher.find()) {
        LogManager.getLogger().warn("Failed to determine Java version; Could not parse {}", javaVersion);
        return -1;
    }

    final String version = matcher.group(1);
    try {
        return Integer.parseInt(version);
    } catch (final NumberFormatException e) {
        LogManager.getLogger().warn("Failed to determine Java version; Could not parse {} from {}", version, javaVersion, e);
        return -1;
    }
}
```